---
title: "Script Allele dosage in Sugarcane"
author: "Fernando S Aguilar"
date: "2023-11-21"
output: pdf_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Libraries
Loading required libraries for the analysis

```{r}
list.of.packages <- c("openxlsx","tidyverse","data.table","fitPoly","gtools")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]                       
if(length(new.packages)) {install.packages(new.packages)}

invisible(lapply(list.of.packages, library, character.only = TRUE))      

rm(list = ls())

```

# Uploading data
Uploading fluorescence ratios for KASP, Allele dosage for Flex-Seq, and GBS+RADSeq
```{r}
# KASP
d=openxlsx::read.xlsx("KASP_Intertek_FluorescenceRatios_185_SNPs_485_Sugarcane_Varieties.xlsx")
d1 <- d[,c("MarkerName","SampleName","ratio")]

#Flex-Seq
TR = fread("Flex_Seq_Allele_dosage_76_SNPs_53_Genotypes.csv", data.table = F)

## GBS+RADSeq
GBRA = fread("GBS_RADSeq_Allele_Dosage_76_SNPs_in_53_sugarcane_varieties_in_Cenicana.csv", data.table = F)
```

#Converting SNPs to KASP
The fluorescence ratios obtained with the KASP technology were converted to a decaploid organism by fitting mixture models, with fitPoly package, to the distribution of fluorescence intensity ratios.

```{r}
#Do not run. It takes long time to solve the mixture models. Instead load the file after this prediction.

# nm= gtools::mixedsort(unique(d1$MarkerName))
# pred=NULL
# for (i in 1:length(nm)) {
#   x=fitOneMarker(ploidy=10, marker=nm[i],data=d)
#   z=x$scores
#   pred=rbind(pred,z)
#   
# }

## Loading the results from the mixture models
pred = read.xlsx("Prediction_Fitpoly_KASP_117_de_185_Markers_KASP_483_Individuos.xlsx")
pred = subset(pred, SampleName %in% TR$Genotype)
pred = subset(pred, MarkerName %in% TR$MarkerName)
pred = pred[,c("MarkerName", "SampleName", "maxgeno")]
pred$maxgeno = pred$maxgeno/10
colnames(pred)[2:3] <- c("Genotype","KASP")
```
Results: Out of the 76 SNP markers, only 42 were succesfully converted to KASP with the fluorescence ratios.

# Correlation between techniques
```{r}
Flex_GBSRADSeq = left_join(TR, GBRA ) 

df_FGR <- Flex_GBSRADSeq %>%
  group_by(MarkerName)%>%
  summarize(r_Flex_GBSRADSeq = cor.test(GBS_RADSeq, Flex_Seq, method = "pearson", na.action = "na.exclude")$estimate,
            
            pval_Flex_GBSRADSeq= cor.test(GBS_RADSeq, Flex_Seq, method = "pearson", na.action = "na.exclude")$p.value,
            
            df_Flex_GBSRADSeq = 
            cor.test(GBS_RADSeq, Flex_Seq, method = "pearson", na.action = "na.exclude")$parameter,
            
            t_Flex_GBSRADSeq = 
            cor.test(GBS_RADSeq, Flex_Seq, method = "pearson", na.action = "na.exclude")$statistic)


Flex_KASP = left_join(pred, TR)

df_FK <- Flex_KASP %>%
  group_by(MarkerName)%>%
  summarize(r_Flex_KASP = cor.test(KASP, Flex_Seq, method = "pearson", na.action = "na.exclude")$estimate,
            
            pval_Flex_KASP = cor.test(KASP, Flex_Seq, method = "pearson", na.action = "na.exclude")$p.value,
            
            df_Flex_KASP = 
            cor.test(KASP, Flex_Seq, method = "pearson", na.action = "na.exclude")$parameter,
            
            t_Flex_KASP = 
            cor.test(KASP, Flex_Seq, method = "pearson", na.action = "na.exclude")$statistic)


GBSRADSeq_KASP = left_join(pred, GBRA)

df_KGR <- GBSRADSeq_KASP %>%
   group_by(MarkerName)%>%
  summarize(r_GBSRADSeq_KASP = cor.test(KASP, GBS_RADSeq, method = "pearson", na.action = "na.exclude")$estimate,
            pval_GBSRADSeq_KASP = cor.test(KASP, GBS_RADSeq, method = "pearson", na.action = "na.exclude")$p.value,
            pval_GBSRADSeq_KASP = cor.test(KASP, GBS_RADSeq, method = "pearson", na.action = "na.exclude")$p.value,
            df_GBSRADSeq_KASP = 
            cor.test(KASP, GBS_RADSeq, method = "pearson", na.action = "na.exclude")$parameter,
            t_GBSRADSeq_KASP = 
            cor.test(KASP, GBS_RADSeq, method = "pearson", na.action = "na.exclude")$statistic)

r_all = left_join(df_FGR, df_KGR) %>% left_join(df_FK)

```

#Heatmap
Creating a heatmap for the correlations obtained between Flex-Seq vs GBS+RADSeq, KASP vs GBS+RADSeq, and KASP vs Flex-Seq
```{r}
df_gbs_flx_kasp <- as.data.frame(r_all)
head(df_gbs_flx_kasp)
df_gbs_flx_kasp_1 <- as.matrix(df_gbs_flx_kasp[,c("MarkerName","r_Flex_GBSRADSeq","r_Flex_KASP","r_GBSRADSeq_KASP")] %>% remove_rownames() %>% column_to_rownames("MarkerName"))
str(df_gbs_flx_kasp_1)

library(pheatmap)

qer = df_gbs_flx_kasp_1

qer[abs(qer) >= 0.8] <- 1
qer[abs(qer) >= 0.5 & abs(qer) < 0.8] <- 0.75
## Flex-Seq vs GBS+RADSeq
qer[abs(qer[,"r_Flex_GBSRADSeq"]) > 0.22 & abs(qer[,"r_Flex_GBSRADSeq"]) < 0.5, "r_Flex_GBSRADSeq"] <- 0.5
qer[abs(qer[,"r_Flex_GBSRADSeq"]) <= 0.22, "r_Flex_GBSRADSeq"] <- 0.25
## KASP vs GBS+RADSeq
qer[abs(qer[,"r_GBSRADSeq_KASP"]) >= 0.276 & abs(qer[,"r_GBSRADSeq_KASP"]) < 0.5,"r_GBSRADSeq_KASP"] <- 0.5
qer[abs(qer[,"r_GBSRADSeq_KASP"]) <= 0.275,"r_GBSRADSeq_KASP"] <- 0.25
## Flex-Seq vs KASP
qer[abs(qer[,"r_Flex_KASP"]) > 0.23 & abs(qer[,"r_Flex_KASP"]) < 0.5,"r_Flex_KASP"] <- 0.5
qer[abs(qer[,"r_Flex_KASP"]) <= 0.23,"r_Flex_KASP"] <- 0.25
## Handling NAs
qer[is.na(qer) ] <- 0

colnames(qer) <- c("Flex-Seq vs GBS+RADSeq","Flex-Seq vs KASP","KASP vs GBS+RADSeq")
qer = qer[,c(1,3,2)]

tiff(filename = "Heatmap_Correlations.tiff", width = 5, height = 10, units = "in", res = 1200)
pheatmap(qer,cluster_rows=FALSE, cluster_cols=FALSE,
   #color=c("grey","green","skyblue","lightblue","red"), 
   legend=T,
   # color = hcl.colors(5, "Oslo"),
   color = c("aliceblue","lightskyblue","steelblue3","dodgerblue3","navy"),
   legend_breaks = c(0.9,0.7,.5,0.3,.1), 
   legend_labels = c("High","Moderate to high","Weak","Non-significant","Missing"), 
   display_numbers = F,
   cellheight = 8, 
   cellwidth = 48, 
   border_color = NA, 
   fontsize = 8,
   angle_col = 315, 
   gaps_col = c(1,2),
   # gaps_row = seq(1,76)
   )

dev.off()


## Transposed
tiff(filename = "Figure_2.tiff", width = 11.5, height = 3, units = "in", res = 1200)
pheatmap(t(qer),cluster_rows=FALSE, cluster_cols=FALSE,
   #color=c("grey","green","skyblue","lightblue","red"), 
   legend=T,
   # color = hcl.colors(5, "Oslo"),
   color = c("aliceblue","lightskyblue","steelblue3","dodgerblue3","navy"),
   legend_breaks = c(0.9,0.7,.5,0.3,.1), 
   legend_labels = c("High","Moderate to high","Weak","Non-significant","Missing"), 
   # legend_labels = c(paste0("High\n", "(r >|0.80|)"),paste0("Moderate to high\n","(|0.50| < r < |0.80|)"),paste0("Weak\n","(|0.23| < r < |0.50|)"),paste0("Non-significant\n","(r < |0.23|)"),"Missing"),
   display_numbers = F,
   cellheight = 48, 
   cellwidth = 8, 
   border_color = NA, 
   fontsize = 8,
   angle_col = 90, 
   # gaps_col = c(1,2),
    gaps_row = seq(1,2)
   )

dev.off()



```

